R/generate_hub_baselines.R

```diff
@@ -0,0 +1,237 @@
+                  ncol = length(args_list$quantile_levels),
+                  byrow = TRUE
+                ),
+                quantile_levels = args_list$quantile_levels
+              )
+            )
+        )
+    })
+  )
+
+  preds_formatted <- preds |>
+    epipredict::flusight_hub_formatter(
+      target = target_name,
+      output_type = "quantile"
+    ) |>
+    tidyr::drop_na(.data$output_type_id) |>
+    dplyr::arrange(.data$target, .data$horizon, .data$location) |>
+    dplyr::select(
+      "reference_date",
+      "horizon",
+      "target",
+      "target_end_date",
+      "location",
+      "output_type",
+      "output_type_id",
+      "value"
+    )
+  return(preds_formatted)
+}
+
+
+#' Generate hub baseline forecasts for a given disease and reference date
+#'
+#' @param base_hub_path Path to the base hub directory.
+#' @param reference_date Reference date (should be a Saturday).
+#' @param disease Disease name ("covid" or "rsv").
+#' @export
+generate_hub_baseline <- function(
+  base_hub_path,
+  reference_date,
+  disease
+) {
+  if (!disease %in% c("covid", "rsv")) {
+    stop("'disease' must be either 'covid' or 'rsv'")
+  }
+  reference_date <- lubridate::as_date(reference_date)
+  desired_max_time_value <- reference_date - 7L
+  dow_supplied <- lubridate::wday(reference_date, week_start = 7, label = FALSE)
+  rng_seed <- as.integer((59460707 + as.numeric(reference_date)) %% 2e9)
+  if (dow_supplied != 7) {
+    cli::cli_abort(
+      message = paste0(
+        "Expected `reference_date` to be a Saturday, day number 7 ",
+        "of the week, given the `week_start` value of Sunday. ",
+        "Got {reference_date}, which is day number ",
+        "{dow_supplied} of the week."
+      )
+    )
+  }
+
+  baseline_model_name <- baseline_model_names[[disease]]
+  output_dirpath <- fs::path(base_hub_path, "model-output", baseline_model_name)
+  if (!fs::dir_exists(output_dirpath)) {
+    fs::dir_create(output_dirpath, recurse = TRUE)
+  }
+
+  hub_target_data <- hubData::connect_target_timeseries(base_hub_path) |>
+    dplyr::collect() |>
+    forecasttools::hub_target_data_as_of()
+
+  preds_hosp <- make_baseline_forecast(
+    target_data = hub_target_data,
+    target_name = glue::glue("wk inc {disease} hosp"),
+    target_label = "Hospital Admissions",
+    reference_date = reference_date,
+    desired_max_time_value = desired_max_time_value,
+    rng_seed = rng_seed
+  )
+
+  preds_ed <- make_baseline_forecast(
+    target_data = hub_target_data,
+    target_name = glue::glue("wk inc {disease} prop ed visits"),
+    target_label = "Proportion ED Visits",
+    reference_date = reference_date,
+    desired_max_time_value = desired_max_time_value,
+    rng_seed = rng_seed
+  )
+
+  forecasttools::write_tabular_file(
+    dplyr::bind_rows(preds_hosp, preds_ed),
+    fs::path(
+      output_dirpath,
+      paste0(as.character(reference_date), "-", baseline_model_name, ".csv")
```

Comment:
Use `glue` here since you're using it elsewhere
```suggestion
      paste0(as.character(reference_date), "-", baseline_model_name, ".csv")
```